extension/ackextension/config.go

```diff
@@ -6,8 +6,30 @@ import (
 	"go.opentelemetry.io/collector/component"
 )
 
+const (
+	defaultMaxNumPartition               = 1_000_000
+	defaultMaxNumPendingAcksPerPartition = 1_000_000
+)
+
 // Config defines configuration for ack extension
 type Config struct {
 	// StorageID defines the storage type of the extension. In-memory type is set by default (if not provided). Future consideration is disk type.
 	StorageID *component.ID `mapstructure:"storage"`
+	// MaxNumPartition Specifies the maximum number of partitions that clients can acquire for this extension instance.
+	// Implementation defines how limit exceeding should be handled.
+	MaxNumPartition uint64 `mapstructure:"max_number_of_partition"`
+	// MaxNumPendingAcksPerPartition Specifies the maximum number of ackIDs and their corresponding status information that are waiting to be queried in each partition.
+	MaxNumPendingAcksPerPartition uint64 `mapstructure:"max_number_of_pending_acks_per_partition"`
+}
+
+func (cfg *Config) Validate() error {
```

Comment:
Can you add tests for the less than 0 cases?

Reply:
since MaxNumPartition and MaxNumPendingAcksPerPartition are of type uint64, the lowest it'll go is 0. I'll add a case for that